middleware.ts

```diff
@@ -0,0 +1,49 @@
+// This Basic Auth code is from https://vancelucas.com/blog/how-to-add-http-basic-auth-to-next-js/
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+
+const adminUser = process.env.ADMIN_USERNAME || "";
+const adminPassword = process.env.ADMIN_PASSWORD || "";
+
+// Step 1. HTTP Basic Auth Middleware for Challenge
+export function middleware(req: NextRequest) {
+  if (!isAuthenticated(req)) {
+    return new NextResponse("Authentication required", {
+      status: 401,
+      headers: { "WWW-Authenticate": "Basic" }
+    });
+  }
+
+  return NextResponse.next();
+}
+
+// Step 2. Check HTTP Basic Auth header if present
+function isAuthenticated(req: NextRequest) {
+  const authheader =
+    req.headers.get("authorization") || req.headers.get("Authorization");
```

Comment:
**Suggestion:** Use 'req.headers.get("authorization")' only, as it is case-insensitive and will match both 'Authorization' and 'authorization'.
```suggestion
  const authheader = req.headers.get("authorization");
```